### PR TITLE
Clean up `initializeQB` action

### PR DESCRIPTION
**Pull request by kulyk**
Adds extensive coverage, refactors, and cleans up query builder's `initializeQB` action. Also converts the action's code to TypeScript, so it's moved to a separate `.ts` file. Also, adds `metabase-lib/mocks` file with question instances factories for unit tests.

### CODE REVIEW

File: frontend/src/metabase/query_builder/actions/core/initializeQB.ts [hunk 1]
```diff
@@ -0,0 +1,337 @@
...
+  }
+
+  if (
+    isSavedCard(card) &&
+    !card?.dataset &&
+    location.pathname?.startsWith("/model")
+  ) {
+    dispatch(setErrorPage(NOT_FOUND_ERROR));
+    return;
+  }
```
Comment (kulyk) [inline frontend/src/metabase/query_builder/actions/core/initializeQB.ts:237]: Before the clean-up, `initializeQB` was not exiting after these checks and was loading query metadata, snippets, etc. without any clear reason. Please LMK if it looks wrong to you to exit the function at this point

File: frontend/src/metabase/query_builder/actions/core/initializeQB.ts [hunk 2]
```diff
@@ -0,0 +1,337 @@
...
+      deserializedCard,
+      originalCard,
+      dispatch,
+      getState,
+    });
+  } else {
+    if (options.metric) {
+      uiControls.isShowingSummarySidebar = true;
+    }
+  }
```
Comment (kulyk) [inline frontend/src/metabase/query_builder/actions/core/initializeQB.ts:250]: @daltojohnso `handleDashboardParameters` performs `verifyMatchingDashcardAndParameters` under the hood. Before the changes this check was non-blocking (ie not `awaited`). I'm not sure if it should actually be blocking?

File: frontend/src/metabase/lib/query/normalize.js [hunk 3]
```diff
@@ -0,0 +1 @@
+export { normalize } from "cljs/metabase.mbql.js";
```
Comment (kulyk) [inline frontend/src/metabase/lib/query/normalize.js:1]: Moved `cljs` import to a JS file as TypeScript can't resolve it when doing a type check

**Review by ranquild — COMMENTED** [review]
I don't think it's possible to actually review this PR. We could either just approve it or split to smaller PRs.

**Comment by kulyk** [comment]
> I don't think it's possible to actually review this PR. We could either just approve it or split to smaller PRs.

Yeah, guess it got too big because of the tests and all the rearrangement. Let me break it down 👍

**Pull request closed without merging**